### DISCUSSION TITLE
Is customizing cell rendering working?

### DISCUSSION THREAD

**Discussion by Prajapati-developer** [mui/toolpad#2148] [Help and support] [answered]
Is customizing cell rendering working? can you give us a brief description of how we should implement it? and also how delete binding will work? and if we want to apply our own css or scss file for custom component then how it will be done?
@Janpot

**Comment by Janpot** [comment]
> Is customizing cell rendering working? can you give us a brief description of how we should implement it?

Just found a regression in the custom columns component selector, adding a fix in https://github.com/mui/mui-toolpad/pull/1840

> and also how delete binding will work?

What do you mean exactly by "delete binding"? You can remove a binding from a property by clicking "remove binding"

> and if we want to apply our own css or scss file for custom component then how it will be done?

Currently we don't support importing css. I can imagine us supporting css modules in the future. Opening a [feature request](https://github.com/mui/mui-toolpad/issues/1841) for that.

**Comment by Prajapati-developer** [comment]
fix in https://github.com/mui/mui-toolpad/pull/1840  is it published as not working? 
delete binding means there is a delete option for datagrid how it is working ?
@Janpot

**Comment by Prajapati-developer** [comment]
and when i am using **Only fetch on manual action** with parameters not working as expected my request is created dynamically but when hitting the button request becomes null so getting the error is there any solution?
@Janpot

**Comment by Janpot** [comment, marked as answer]
> delete binding means there is a delete option for datagrid how it is working ?

It looks like that's a leftover from an earlier experiment on providing richer connection to tabular data. Removing it again https://github.com/mui/mui-toolpad/pull/1850. We're tracking further exploration of the topic in https://github.com/mui/mui-toolpad/issues/385

For the time being we recommend binding a manual query parameter to the `dataGrid.selection` and triggering it on button click.

> and when i am using Only fetch on manual action with parameters not working as expected my request is created dynamically but when hitting the button request becomes null so getting the error is there any solution?

Two ways to debug these at the moment is
- bind a `text` compenent to the `query.error.message` state, then click the button
- in the button event handler `console.log(await query.fetch())` should give you some feedback as well in the browser console.

Granted, we need to improve debuggability of the network layer. Adding an umbrella issue to keep track of that https://github.com/mui/mui-toolpad/issues/1851
